tests/lib/compose_features.py

```diff
@@ -0,0 +1,273 @@
+#!/usr/bin/env python3
+
+import argparse
+import json
+import os
+import shutil
+from typing import Any
+import feature_dict
+
+
+def _parse_file_name(file_name: str) -> tuple[str, str, str, str]:
+    '''
+    Given a file name in the format with double slashes <backend>:<system>:suite--path--task:variant
+    and optionally an extension, it returns the original name, the suite name, the task name, 
+    and the variant name. So in the example, it returns:
+    - original_name = <backend>:<system>:suite/path/task:variant
+    - suite_name = suite/path
+    - task_name = task
+    - variant_name = variant
+
+    :param file_name: The file name to parse
+    :returns: A tuple with the original name, the suite name, the task name and the variant name. If variant is not present, it returns None.
+    '''
+    file_name = os.path.splitext(file_name)[0]
+    original_name = file_name.replace('--', '/')
+    task = ':'.join(original_name.split(':')[2:])
+    suite_name = '/'.join(task.split('/')[:-1])
+    task_name = task.split('/')[-1]
+    variant_name = ''
+    if task_name.count(':') == 1:
+        variant_name = task_name.split(':')[1]
+        task_name = task_name.split(':')[0]
+    return original_name, suite_name, task_name, variant_name
+
+
+def _compose_test(dir: str, file: str, failed_tests: str) -> feature_dict.TaskFeatures:
+    '''
+    Creates a dictionary with the features of a test and test information.
+    The features are read from the file and the test information is extracted from the file name.
+
+    :param dir: The directory where the file is located
+    :param file: The file name
+    :param failed_tests: A list of failed tests
+    :returns: A dictionary with test information and features
+    '''
+    with open(os.path.join(dir, file), 'r') as f:
+        original, suite_name, result_name, variant_name = _parse_file_name(
+            file)
+        features = feature_dict.TaskFeatures(
+            suite=suite_name,
+            task_name=result_name,
+            variant=variant_name,
+            success=original not in failed_tests
+        )
+        features.update(json.loads(f.read()))
+        return features
+
+
+def _compose_env_variables(env_variables: str) -> list[feature_dict.EnvVariables]:
+    '''
+    Given environment variables in the form of a comma-separated list of key=value,
+    it creates a list of dictionaries of [{"name": <env1-name>, "value": <env1-value>}...]
+
+    :param env_variables: a comma-seprated list of key=value environment variables
+    :returns: A list of dictionaries
+    '''
+    composed = []
+    for env in env_variables.split(',') if env_variables else []:
+        name, value = env.split('=')
```

Comment:
what about foo=bar=froz?

I think you want `env.partition('=')` which behaves like `strings.Cut` in Go.